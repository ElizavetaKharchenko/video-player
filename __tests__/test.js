import { promises as fs } from 'fs';
import path from 'path';
import init from '../src/app';

const getHtml = () => document.documentElement.innerHTML;

beforeEach(async () => {
  const pathToHtml = path.resolve(__dirname, '__fixtures__/index.html');
  const html = await fs.readFile(pathToHtml, 'utf8');
  document.body.innerHTML = html;

  init();
});

test('start player', () => {
  expect(getHtml()).toMatchSnapshot();
});

test('overlay play button hidden on play', () => {
  const video = document.getElementById('mainVideo');
  video.dispatchEvent(new window.Event('play'));
  expect(getHtml()).toMatchSnapshot();
});

test('overlay play button show on stop', () => {
  const video = document.getElementById('mainVideo');
  video.dispatchEvent(new window.Event('pause'));
  expect(getHtml()).toMatchSnapshot();
});

test('overlay reload button show on end', () => {
  const video = document.getElementById('mainVideo');
  video.dispatchEvent(new window.Event('ended'));
  expect(getHtml()).toMatchSnapshot();
});

test('sound', () => {
  const dynamic = document.getElementById('dynamic');
  dynamic.click();
  expect(getHtml()).toMatchSnapshot();
});
