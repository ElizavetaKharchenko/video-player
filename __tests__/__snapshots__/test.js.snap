// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlay play button hidden on play 1`] = `
"<head></head><body>

  
    <title>Video-player</title>
  
  
    <div class=\\"container\\">
      <div class=\\"video-container\\">
        <video id=\\"mainVideo\\" width=\\"768\\" height=\\"432\\" muted=\\"\\" autoplay=\\"\\">
          <source src=\\"mainVideo.mp4\\" type=\\"video/mp4;codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">
          <source src=\\"mainVideo.webm\\" type=\\"video/webm; codecs=&quot;vorbis,vp8&quot;\\">
        </video>
        <div class=\\"overlay-btn-play\\">
          <i id=\\"overlay-btn-play\\" class=\\"fas fa-play-circle fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
        <div class=\\"overlay-btn-reload\\">
          <i id=\\"overlay-btn-reload\\" class=\\"fas fa-sync-alt fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
      </div>
      <div id=\\"controls\\">
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"playPause\\" class=\\"fas fa-stop\\"></i></a>
        <div class=\\"progress\\">
          <div class=\\"progress-bar\\"></div>
        </div>
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"dynamic\\" class=\\"fas fa-volume-off\\"></i></a>
      </div>
    </div>
  
</body>"
`;

exports[`overlay play button show on stop 1`] = `
"<head></head><body>

  
    <title>Video-player</title>
  
  
    <div class=\\"container\\">
      <div class=\\"video-container\\">
        <video id=\\"mainVideo\\" width=\\"768\\" height=\\"432\\" muted=\\"\\" autoplay=\\"\\">
          <source src=\\"mainVideo.mp4\\" type=\\"video/mp4;codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">
          <source src=\\"mainVideo.webm\\" type=\\"video/webm; codecs=&quot;vorbis,vp8&quot;\\">
        </video>
        <div class=\\"overlay-btn-play\\">
          <i id=\\"overlay-btn-play\\" class=\\"fas fa-play-circle fa-5x\\" style=\\"visibility: visible;\\"></i>
        </div>
        <div class=\\"overlay-btn-reload\\">
          <i id=\\"overlay-btn-reload\\" class=\\"fas fa-sync-alt fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
      </div>
      <div id=\\"controls\\">
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"playPause\\" class=\\"fas fa-play-circle\\"></i></a>
        <div class=\\"progress\\">
          <div class=\\"progress-bar\\"></div>
        </div>
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"dynamic\\" class=\\"fas fa-volume-off\\"></i></a>
      </div>
    </div>
  
</body>"
`;

exports[`overlay reload button show on end 1`] = `
"<head></head><body>

  
    <title>Video-player</title>
  
  
    <div class=\\"container\\">
      <div class=\\"video-container\\">
        <video id=\\"mainVideo\\" width=\\"768\\" height=\\"432\\" muted=\\"\\" autoplay=\\"\\">
          <source src=\\"mainVideo.mp4\\" type=\\"video/mp4;codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">
          <source src=\\"mainVideo.webm\\" type=\\"video/webm; codecs=&quot;vorbis,vp8&quot;\\">
        </video>
        <div class=\\"overlay-btn-play\\">
          <i id=\\"overlay-btn-play\\" class=\\"fas fa-play-circle fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
        <div class=\\"overlay-btn-reload\\">
          <i id=\\"overlay-btn-reload\\" class=\\"fas fa-sync-alt fa-5x\\" style=\\"visibility: visible;\\"></i>
        </div>
      </div>
      <div id=\\"controls\\">
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"playPause\\" class=\\"fas fa-play-circle\\"></i></a>
        <div class=\\"progress\\">
          <div class=\\"progress-bar\\"></div>
        </div>
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"dynamic\\" class=\\"fas fa-volume-off\\"></i></a>
      </div>
    </div>
  
</body>"
`;

exports[`sound 1`] = `
"<head></head><body>

  
    <title>Video-player</title>
  
  
    <div class=\\"container\\">
      <div class=\\"video-container\\">
        <video id=\\"mainVideo\\" width=\\"768\\" height=\\"432\\" muted=\\"\\" autoplay=\\"\\">
          <source src=\\"mainVideo.mp4\\" type=\\"video/mp4;codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">
          <source src=\\"mainVideo.webm\\" type=\\"video/webm; codecs=&quot;vorbis,vp8&quot;\\">
        </video>
        <div class=\\"overlay-btn-play\\">
          <i id=\\"overlay-btn-play\\" class=\\"fas fa-play-circle fa-5x\\"></i>
        </div>
        <div class=\\"overlay-btn-reload\\">
          <i id=\\"overlay-btn-reload\\" class=\\"fas fa-sync-alt fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
      </div>
      <div id=\\"controls\\">
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"playPause\\" class=\\"fas fa-play-circle\\"></i></a>
        <div class=\\"progress\\">
          <div class=\\"progress-bar\\"></div>
        </div>
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"dynamic\\" class=\\"fas fa-volume-up\\"></i></a>
      </div>
    </div>
  
</body>"
`;

exports[`start player 1`] = `
"<head></head><body>

  
    <title>Video-player</title>
  
  
    <div class=\\"container\\">
      <div class=\\"video-container\\">
        <video id=\\"mainVideo\\" width=\\"768\\" height=\\"432\\" muted=\\"\\" autoplay=\\"\\">
          <source src=\\"mainVideo.mp4\\" type=\\"video/mp4;codecs=&quot;avc1.42E01E, mp4a.40.2&quot;\\">
          <source src=\\"mainVideo.webm\\" type=\\"video/webm; codecs=&quot;vorbis,vp8&quot;\\">
        </video>
        <div class=\\"overlay-btn-play\\">
          <i id=\\"overlay-btn-play\\" class=\\"fas fa-play-circle fa-5x\\"></i>
        </div>
        <div class=\\"overlay-btn-reload\\">
          <i id=\\"overlay-btn-reload\\" class=\\"fas fa-sync-alt fa-5x\\" style=\\"visibility: hidden;\\"></i>
        </div>
      </div>
      <div id=\\"controls\\">
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"playPause\\" class=\\"fas fa-play-circle\\"></i></a>
        <div class=\\"progress\\">
          <div class=\\"progress-bar\\"></div>
        </div>
        <a class=\\"btn\\" href=\\"#\\"><i id=\\"dynamic\\" class=\\"fas fa-volume-off\\"></i></a>
      </div>
    </div>
  
</body>"
`;
